### PR TITLE
Add expression conversion, getters, tests

### PR DESCRIPTION
**Pull request by tobrun**
Closes https://github.com/mapbox/mapbox-gl-native/issues/10721, https://github.com/mapbox/mapbox-gl-native/issues/10722, https://github.com/mapbox/mapbox-gl-native/issues/11247, https://github.com/mapbox/mapbox-gl-native/issues/10868

This PR adds accessors for expressions and removes the old API for doing DDS. 
 - [x] add conversion template to convert mapbox::geometry::value to JsonArray
 - [x] convert JsonArray.java to Expression.java
 - [x] add basic set/get tests for camera, source and composite functions
 - [x] add edge case test for internal expression conversion (eg. rgb becomes rgba)
 - [x] remove old DDS stop concepts in c++ and java
 - [x] replace all old DDS examples in test app with expressions
 - [x] replace anonymous inner classes by lamba constructs in generated test code https://github.com/mapbox/mapbox-gl-native/issues/10868
 - [x] simplify function concept in java code (eg. only expose expression on PropertyValue.java) 
 - [x] add generated tests for using an expression in any paint/layout property 
 - [x] cleanup code

### CODE REVIEW

File: platform/android/src/style/conversion/property_value.hpp [hunk 1]
```diff
@@ -39,7 +40,7 @@ class PropertyValueEvaluator {
     }
 
     jni::jobject* operator()(const mbgl::style::CompositeFunction<T> &value) const {
-      return *convert<jni::jobject*, mbgl::style::CompositeFunction<T>>(env, value);
+        return *convert<jni::jobject*, mbgl::style::CompositeFunction<T>>(env, value);
```
[the remark this answers is not in the document]
Comment (tobrun) [reply platform/android/src/style/conversion/property_value.hpp]: this one was intentional 😄there was a space missing, see https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/src/style/conversion/property_value.hpp#L41
